### ISSUE TITLE
hide page icons in menu when user don't have permission to see

### ISSUE THREAD

**Issue by AlessandroAmos** [lunarphp/lunar#740]
**Is your feature request related to a problem? Please describe.**
in my opinion there is no need to display the pages that the user does not have access to in the menu

**Describe the solution you'd like**
just hide the icons of the pages that the user does not have permission to see

**Comment by markmead** [comment]
This could definitely be handled better. However, I don't believe the menu item should be removed entirely.

If that were the case a user would not know the areas of the admin hub and there may be a time that they need access to an area, but won't know that area exists.

**Comment by glennjacobs** [comment]
> This could definitely be handled better. However, I don't believe the menu item should be removed entirely.
> 
> If that were the case a user would not know the areas of the admin hub and there may be a time that they need access to an area, but won't know that area exists.

I don't agree. If you don't have permission to something, you shouldn't have a menu item showing. 

Admins will always seem all menu items.

**Comment by markmead** [comment]
If I was updating products and then thought "Oh, I need to update collections" I don't want to see an empty menu, I want to see "Collections" and then clicking that informs me I do not have access but I can request it.

Otherwise, I'm clueless as to what the system can/can't do.

**Comment by illumws** [comment]
> If I was updating products and then thought "Oh, I need to update collections" I don't want to see an empty menu, I want to see "Collections" and then clicking that informs me I do not have access but I can request it.
> 
> Otherwise, I'm clueless as to what the system can/can't do.

but the point is that if you don't have permission to access the collections page, you shouldn't even try, because your workflow doesn't include collections registration.

**Comment by glennjacobs** [comment]
> If I was updating products and then thought "Oh, I need to update collections" I don't want to see an empty menu, I want to see "Collections" and then clicking that informs me I do not have access but I can request it.
> 
> Otherwise, I'm clueless as to what the system can/can't do.

I've never seen such functionality in any other systems. It should be hidden.

But generally, it's someone who admins the catalogue but doesn't have access to Orders or similar. Having access to Products but not Collections is an unlikely scenario.

**Comment by glennjacobs** [comment]
> Shopify does it and that's by far the best admin experience I've had.

I'm happy to learn, but I feel it would be very frustrating for a user to keep clicking on things and then to see "access denied" and I'm sure devs would want the control in Lunar as we're more likely to have custom menu items which may be sensitive.

How do Shopify prevent it from being annoying?

**Comment by AlessandroAmos** [comment]
just disable the buttons 
![image](https://user-images.githubusercontent.com/118480471/203066741-ce10aa6f-047e-4cd9-9137-a7d6f9a7fda1.png)

**Comment by markmead** [comment]
Oh, so you're interested now? 😉 

Shopify shows the menu item but with a lower opacity, when this is clicked it takes you to that URL but shows a panel that allows you to request access. This sends an email to the admin of the store.

Interestingly, on the store I grabbed the screenshots from the menu items weren't clickable. This could be the new approach Shopify is taking as they were clickable on Saturday as I used it to request access... 🤔

![](https://user-images.githubusercontent.com/50486078/203067172-9d54e3ca-0d82-4aba-8044-6efeaeec4404.png)
![](https://user-images.githubusercontent.com/50486078/203067174-443f5a30-c143-4c44-8bf8-b1c2076fe8c5.png)
![](https://user-images.githubusercontent.com/50486078/203067167-9206f1ef-104c-461f-a2a1-d2f6f3948259.png)

**Comment by glennjacobs** [comment]
@alecritson what are your thoughts on this? I can see pros and cons for both approaches.

**Comment by markmead** [comment]
OK, so Shopify has made a change so that the menu items you don't have access to are not clickable. You now have to contact the admin to request access.

It's strange that I've been able to do this, I think it's fried Reyse from Shopify supports brain... 😅

**Comment by alecritson** [comment]
@glennjacobs I can see the argument for both cases. Personally, if I log into a system and all I can see is one menu item or something, it just looks really sparse. 

Just thinking down the line this might not be limited to just menu items, if actions require permissions and a user is left trying to figure out how you perform a certain action when it's hidden could be more frustrating?

Also by showing everything it helps with learning of the system as you can see everything that's available, even if you can't access it.

There's certainly been times where we've been on a call with screen sharing and it can be quite jarring if you're clicking through bits and we're sitting here like "I can't see any of this, I have no idea where these things are"

On the flip side, yeah you could argue you shouldn't be able to see what you don't have access to, someone might not even want to see anything they don't have access to as it doesn't concern them.

If in doubt, add a config option "Hide restricted actions & pages" 😅

**Comment by glennjacobs** [comment]
Good old config!

Another thought is, disabled menu items may get in the way. E.g. pushing the menu item you do have access to down or w.e.

Rather than config, could this be a user account option? E.g. "Hide disabled options".

**Comment by glennjacobs** [comment]
@alecritson we have a page where you can edit your profile I think? So for now could be a simple option. Down the line we could expend it out into a proper user settings area.

**Comment by alecritson** [comment]
@glennjacobs The profile editing doesn't do anything that's outside of what Laravel provides on the users table. We'd have to add a new column which is something I thought we wanted to avoid.

**Comment by glennjacobs** [comment]
@alecritson I guess we could store in a one-to-one table, but yeah, opening a can of worms potentially here.
